### PR TITLE
[Enhancement] - User can successfully change Email

### PR DESCRIPTION
**Pull request by Rcureton**
## What
- User can successfully change email. The `UI` is accurate if you switch to production the Gif is on staging.
- Created an extension function for setting the custom `Snackbar` background.
- Merged in changes from `master` and a prior branch hence all the file changes.

## Story
- [User can update email success](https://trello.com/c/HTi8ztJ9/883-email-update-success)

## See 👀
change_email

### CODE REVIEW

File: app/src/main/java/com/kickstarter/extensions/SnackbarExt.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import android.view.ViewGroup
+import com.kickstarter.R
+
+fun Snackbar.success(context: Context) {
+    val params = this.view.layoutParams as ViewGroup.MarginLayoutParams
+    params.setMargins(12, 12, 12, 12)
+    this.view.layoutParams = params
+    this.view.background = ContextCompat.getDrawable(context, R.drawable.change_email_success_bg)
+    ViewCompat.setElevation(this.view, 6f)
+}
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/extensions/SnackbarExt.kt]: nl

File: app/src/main/java/com/kickstarter/extensions/SnackbarExt.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import android.content.Context
+import android.support.design.widget.Snackbar
+import android.support.v4.content.ContextCompat
+import android.support.v4.view.ViewCompat
+import android.view.ViewGroup
+import com.kickstarter.R
+
+fun Snackbar.success(context: Context) {
+    val params = this.view.layoutParams as ViewGroup.MarginLayoutParams
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/extensions/SnackbarExt.kt]: Magic numbers make code harder to understand.
Let's try something like this instead:
```   
    val grid1 = context.resources.getDimensionPixelSize(R.dimen.grid_1)
    val grid2 = context.resources.getDimensionPixelSize(R.dimen.grid_2)
    params.setMargins(grid1, 0, grid1, grid2)
```

File: app/src/main/java/com/kickstarter/viewmodels/ChangeEmailViewModel.kt [hunk 3]
```diff
@@ -12,6 +17,9 @@ import rx.subjects.PublishSubject
 interface ChangeEmailViewModel {
 
     interface Inputs {
+        /** Call when the make network call button has been clicked.  */
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ChangeEmailViewModel.kt]: This looks like copy pasta 👀 🍝

File: app/src/main/java/com/kickstarter/viewmodels/ChangeEmailViewModel.kt [hunk 4]
```diff
@@ -20,11 +28,11 @@ interface ChangeEmailViewModel {
         /** Emits the logged in user's email address.  */
         fun email(): Observable<String>
 
-        /** Emits the logged in user's name.  */
-        fun name(): Observable<String>
-
         /** Emits a boolean that determines if a network call is in progress.  */
         fun showProgressBar(): Observable<Boolean>
+
+        /** Emits the logged in user's email which we use get the success of the mutation. */
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ChangeEmailViewModel.kt]: Are you doing anything with this value?

File: app/src/main/res/drawable/change_email_success_bg.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/ksr_cobalt_500" />
+  <corners android:radius="@dimen/grid_1" />
+</shape>
```
Comment (eoji) [inline app/src/main/res/drawable/change_email_success_bg.xml]: nl

File: app/src/test/java/com/kickstarter/viewmodels/ChangeEmailViewModelTest.kt [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import com.kickstarter.KSRobolectricTestCase
+import com.kickstarter.libs.Environment
+import com.kickstarter.mock.services.MockApolloClient
+import org.junit.Test
+import rx.Observable
+import rx.observers.TestSubscriber
+
+class ChangeEmailViewModelTest : KSRobolectricTestCase() {
+
+    private lateinit var vm: TestApolloViewModel.ViewModel
```
Comment (eoji) [inline app/src/test/java/com/kickstarter/viewmodels/ChangeEmailViewModelTest.kt]: Shouldn't this be `ChangeEmailViewModel.ViewModel`?

**Comment by Rcureton** [comment]
[Email Update: Success](https://trello.com/c/HTi8ztJ9/883-email-update-success)

**Pull request merged**